api/types.proto

```diff
@@ -713,3 +713,34 @@ message ManagerStatus {
 	// Reachability specifies whether this node is reachable.
 	RaftMemberStatus.Reachability reachability = 4;
 }
+
+// SecretType provides information about what kind of secret this is
+enum SecretType {
+	// CONTAINER secrets are user-specified secrets that should be injected
+	// into the container as an environment variable or file.
+	CONTAINER = 0 [(gogoproto.enumvalue_customname) = "ContainerSecret"];
```

Comment:
After some discussion, it seems we can get rid of the type altogether and replace it with an `internal` bool field. For the most part, we seem to only need to hide this for network secrets. In the future, we could introduce "scope" or "namespace" if we want to add dimensions.